tests/server/test_auth.py

```diff
@@ -606,3 +604,291 @@ async def test_get_input_for_policy_engine(capture_input_for_policy_engine: Capt
     assert pe_input["input"]["token"]["urn:inmanta:ct"] == ["api"]
     assert pe_input["input"]["token"][const.INMANTA_ROLES_URN] == {env_id: "test"}
     assert pe_input["input"]["token"][const.INMANTA_IS_ADMIN_URN] is False
+
+
+@pytest.mark.parametrize(
+    "access_policy",
+    [
+        """
+        package policy
+
+        # Write the information about the endpoint into a variable
+        # to make the policy easier to read.
+        endpoint_data := data.endpoints[input.request.endpoint_id]
+
+        # The environment used in the request
+        request_environment := input.request.parameters[endpoint_data.environment_param] if {
+            endpoint_data.environment_param != null
+        } else := null
+
+        default allow := false
+
+        # Allow access if the user has the role a_role.
+        allow if {
+            input.token["urn:inmanta:roles"][request_environment] == "a_role"
+        }
+
+        # Allow access to admin user.
+        allow if {
+            input.token["urn:inmanta:is_admin"]
```

Comment:
admin is not a role, but a special case? I suppose that is a good thing.

Reply:
Yes. It's a marker to indicate you are a global admin, which can access everything. It's not scoped to an environment.